web/packages/teleport/src/Assist/Chat/Chat.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import styled, { keyframes } from 'styled-components';
+
+import { Dots } from 'teleport/Assist/Dots';
+
+import { useMessages } from '../contexts/messages';
+
+import { Author, Type } from '../services/messages';
+
+import { ChatBox } from './ChatBox';
+import { ChatItem } from './ChatItem';
+import { ExampleChatItem } from './ChatItem/ChatItem';
+
+const Container = styled.div`
+  flex: 1;
+  position: relative;
+  background: #222c5a;
+  overflow: hidden;
+  display: flex;
+  margin: 40px;
+  border-radius: 20px;
+  flex-direction: column;
+  border: 1px solid rgba(255, 255, 255, 0.1);
+  box-shadow: 0 10px 30px rgba(0, 0, 0, 0.2);
+`;
+
+const Header = styled.div`
+  padding: 30px;
+  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
+  box-shadow: 0 1px 0 rgba(255, 255, 255, 0.2);
+  color: white;
+  position: relative;
+  z-index: 100;
+  font-size: 22px;
+  font-weight: bold;
+`;
+
+const Content = styled.div`
+  flex: 1 1 auto;
+  overflow-y: auto;
+  min-height: 100px;
+`;
+
+const Padding = styled.div`
+  padding: 30px;
+  box-sizing: border-box;
+`;
+
+const appear = keyframes`
+  0% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: 1;
+  }
+`;
+
+const LoadingContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  opacity: 0;
+  margin-top: 50px;
+  animation: ${appear} 0.6s linear forwards;
+  animation-delay: 1s;
+`;
+
+export function Chat() {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const { send, messages, streaming } = useMessages();
+
+  const scrollTextarea = useCallback(() => {
+    ref.current?.scrollIntoView({ behavior: 'smooth' });
+  }, [ref.current]);
+
+  useEffect(() => {
+    scrollTextarea();
+  }, [messages, scrollTextarea]);
+
+  const handleSubmit = useCallback(
+    (message: string) => {
+      send({
+        author: Author.User,
```

Comment:
I don't think we need to specify this here. We'll handle it in the backend so you can't go bananas with it. And you shouldn't be able to send as anything other than yourself either.

Reply:
Yeah, this is just left over from my backend. It needs to change once the spec for what needs to go to the backend is decided on.

Reply:
I've changed this to just publish the string of the message to the websocket